sklearn/svm/classes.py

```diff
@@ -1128,3 +1130,169 @@ def predict(self, X):
         """
         y = super(OneClassSVM, self).predict(X)
         return np.asarray(y, dtype=np.intp)
+
+
+class SVDD(BaseLibSVM):
+    """Support Vector Data Description (SVDD) for Unsupervised Outlier Detection.
+
+    Estimate the support of a high-dimensional distribution by finding the
+    tightest soft boundary hypersphere around a data set, which permits at
+    most the fraction ``nu`` (``0 < nu <= 1``) of the data as outliers.
+
+    The implementation is based on libsvm.
+
+    Read more in the :ref:`User Guide <svm_outlier_detection>`.
+
+    Parameters
+    ----------
+    kernel : string, optional (default='rbf')
+         Specifies the kernel type to be used in the algorithm.
+         It must be one of 'linear', 'poly', 'rbf', 'sigmoid', 'precomputed'
+         or a callable.
+         If none is given, 'rbf' will be used. If a callable is given it is
+         used to precompute the kernel matrix.
+
+    nu : float, optional
+        An upper bound on the fraction of training errors and a lower bound
+        of the fraction of support vectors. Should be in the interval (0, 1].
+        By default 0.5 will be taken.
+
+    degree : int, optional (default=3)
+        Degree of the polynomial kernel function ('poly').
+        Ignored by all other kernels.
+
+    gamma : float, optional (default='auto')
+        Kernel coefficient for 'rbf', 'poly' and 'sigmoid'.
+        If gamma is 'auto' then 1/n_features will be used instead.
+
+    coef0 : float, optional (default=0.0)
+        Independent term in kernel function.
+        It is only significant in 'poly' and 'sigmoid'.
+
+    tol : float, optional
+        Tolerance for stopping criterion.
+
+    shrinking : boolean, optional
+        Whether to use the shrinking heuristic.
+
+    cache_size : float, optional
+        Specify the size of the kernel cache (in MB).
+
+    verbose : bool, default: False
+        Enable verbose output. Note that this setting takes advantage of a
+        per-process runtime setting in libsvm that, if enabled, may not work
+        properly in a multithreaded context.
+
+    max_iter : int, optional (default=-1)
+        Hard limit on iterations within solver, or -1 for no limit.
+
+    random_state : int, RandomState instance or None, optional (default=None)
```

Comment:
we should remove `random_state` as the implementation is not random (this parameter has been recently deprecated for the OneClassSVM).